torch/autograd/profiler.py

```diff
@@ -602,12 +602,17 @@ def __init__(self, name: str, args: Optional[str] = None):
         )
 
     def __enter__(self):
-        self.record = torch.ops.profiler._record_function_enter_new(
-            self.name, self.args
-        )
+        from torch.fx.experimental.proxy_tensor import maybe_disable_fake_tensor_mode
+
+        with torch.utils._python_dispatch._disable_current_modes(), maybe_disable_fake_tensor_mode():
```

Comment:
From my experience in touching this code a long time ago, doing anything but the bare minimum to call a single c++ op here is degrading performance significantly (specific use case for Meta employees is [here](https://fb.workplace.com/groups/1069285536500339/permalink/5027037044058482/))

Reply:
Should we just special case this operator then?

Reply:
@ydwu4 why did we need additional handling around this?

Reply:
`_record_function_enter_new, _record_function_exit._RecordFunction` defined one implementation for all [backends](https://github.com/pytorch/pytorch/blob/main/torch/csrc/autograd/record_function_ops.cpp#L129-L140). 

[dynamo_timed](https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/utils.py#L226-L273) used them in inductor to profile the compilation time and it's on the critical path. See the [test failure in previous version](https://hud.pytorch.org/pytorch/pytorch/pull/120045?sha=caa2d677aa22c37591f5d9d9edbaaf797314292b) if we don't fakify the inputs . 

If we fakify the input to _RecordFunction, the profiling is not executed. I believe that these profiler methods can be used to profile fake tensor propagation performance, so I decided to keep their behavior unchanged. However, adding maybe_disable_fake_tensor_mode before calling the op might incur too much overhead, according to Alban. Therefore, I came up with an additional handling approach.

For custom ops that have one implementation for all backends, should we fakify the TorchScriptObject inputs if running under fake tensor mode? My current thinking is that we should only fakify those inputs that users want to trace as a node in the graph module. For the rest, we should keep their original behavior since these operators indeed want to execute the real operation under fake mode. What do you think?